vortex-array/src/compute/take.rs

```diff
@@ -1,13 +1,21 @@
 use log::info;
+use vortex_dtype::PType::{U32, U64};
+use vortex_dtype::{DType, Nullability};
 use vortex_error::{vortex_err, VortexResult};
 
-use crate::{Array, IntoArray, IntoCanonical};
+use crate::{Array, ArrayDType, IntoArray, IntoCanonical};
 
 pub trait TakeFn {
     fn take(&self, indices: &Array) -> VortexResult<Array>;
 }
 
 pub fn take(array: &Array, indices: &Array) -> VortexResult<Array> {
+    assert!(
+        indices.dtype() == &DType::Primitive(U64, Nullability::NonNullable)
+            || indices.dtype() == &DType::Primitive(U32, Nullability::NonNullable),
+        "indices must be non-nullable u32 or u64"
```

Comment:
There's a `is_unsigned_int` method on `PType`, so if you use `matches!` you can write this more compactly

Reply:
thanks, but instead all (positive) ints (including signed) should be supported, I guess I could add that instead?